xbmc/addons/PVRClient.cpp

```diff
@@ -2093,3 +2093,42 @@ void CPVRClient::OnPowerSavingDeactivated(void)
   }
   catch (std::exception &e) { LogException(e, __FUNCTION__); }
 }
+
+PVRError CPVRClient::TranslateError(PVR_ERROR error)
+{
+  PVRError error;
+  switch (error)
+  {
+  case PVR_ERROR_NO_ERROR:
+    error = PVRError_NO_ERROR;
+    break;
+  case PVR_ERROR_NOT_IMPLEMENTED:
+    error = PVRError_NOT_IMPLEMENTED;
+    break;
+  case PVR_ERROR_SERVER_ERROR:
+    error = PVRError_SERVER_ERROR;
+    break;
+  case PVR_ERROR_SERVER_TIMEOUT:
+    error = PVRError_SERVER_TIMEOUT;
+    break;
+  case PVR_ERROR_REJECTED:
+    error = PVRError_REJECTED;
+    break;
+  case PVR_ERROR_ALREADY_PRESENT:
+    error = PVRError_ALREADY_PRESENT;
+    break;
+  case PVR_ERROR_INVALID_PARAMETERS:
+    error = PVRError_INVALID_PARAMETERS;
+    break;
+  case PVR_ERROR_RECORDING_RUNNING:
+    error = PVRError_RECORDING_RUNNING;
+    break;
+  case PVR_ERROR_FAILED:
+    error = PVRError_FAILED;
+    break;
+  case PVR_ERROR_UNKNOWN:
+  default:
+    error = PVRError_UNKNOWN;
+  }
+  return error;
+}
```

Comment:
Has added a way as example to bring error codes from add-on headers to a one used on Kodi.

As long it is only one CPVRClient class present looks it crazy ;)

To use different levels comes on end this on `CPVRClientBase` and the the level class looks like:
```cpp
namespace V3
{
namespace KodiAPI
{
class CPVRClient
  : public CPVRClientBase
```